src/Runner.Listener/SelfUpdater.cs

```diff
@@ -59,19 +66,44 @@ public async Task<bool> SelfUpdate(AgentRefreshMessage updateMessage, IJobDispat
             {
                 var totalUpdateTime = Stopwatch.StartNew();
 
+                // Copy dotnet runtime and externals of current runner to a temp folder
+                // So we can re-use them with trimmed runner package, if possible.
+                // This process is best effort, if we can't use trimmed runner package, 
+                // we will just go with the full package.
+                var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(token);
+                _cloneAndCalculateContentHashTask = CloneAndCalculateAssetsHash(_dotnetRuntimeCloneDirectory, _externalsCloneDirectory, linkedTokenSource.Token);
```

Comment:
nit: I'm wondering if the time saved here is worth doing before we know if we should copy these or not. Typically disk copy speeds are much faster than downloads, are we actually saving time?

Reply:
The reason I start the copy is we might run into errors when copying or computing the hash.
So I will just fall back to the full packages early if anything went wrong.